apps/docs/src/content/reference/extras/instanced-sprite.mdx

```diff
@@ -0,0 +1,454 @@
+            type: 'number',
+            required: false,
+            description: `Sets an exact frame to display. If animationName is set, it's the id of the frame in that given animation, otherwise it's a spritesheet frame id.`
+          }
+        ]
+    }}
+	/>
+
+
+### binding
+
+The `InstancedSpriteMesh` class is the foundation behind the `<InstancedSprite>` component, written to enable efficient instancing of animated sprites within a
+Three.js environment. The `<InstancedSprite>` component lets you bind to it through `ref`.
+
+The class extends the capabilities of the [troika's InstancedUniformsMesh](https://protectwise.github.io/troika/three-instanced-uniforms-mesh/). For an in-depth exploration of `InstancedSpriteMesh` and its features, refer to the documentation available at [InstancedSpriteMesh docs](https://three-kit.vercel.app/instancedsprite/01-instanced-sprite-mesh/).
+
+
+## Spritesheets
+
+### SpritesheetMetadata
+
+Object used in `buildSpritesheet` function has to be compliant with the `SpritesheetMetadata` type format. This type is structured to
+accommodate the metadata for one or multiple sprite files within a single spritesheet.
+
+```ts
+type SpritesheetMetadata = {
+  url: string
+  type: 'rowColumn' | 'frameSize'
+  width: number
+  height: number
+  animations: {
+    name: string
+    frameRange: [number, number]
+  }[]
+}[]
+
+```
+
+#### Understanding SpritesheetMetadata
+
+A `SpritesheetMetadata` is an array, with each entry representing metadata fields for one sprite:
+
+	- `url`: Specifies the path or URL to the sprite image file.
+	- `type`: Determines the method of defining the spritesheet dimensions. Type `"rowColumn"` specifies the layout in terms of rows
+	and columns within the image, and type `"frameSize"` instead defines the size of each frame, allowing the utility to calculate the layout.
+	- `width` and `height`: Depending on type, these refer to the number of `columns` and `rows` (`"rowColumn"`) or the dimensions of a single frame (`"frameSize"`).
+	- `animations`: An array detailing the animations, where each animation has a name and a `frameRange`. The `frameRange` is a tuple
+	marking the start and end frames of the animation.
+
+#### Typesafety
+
+For improved developer experience when working with TypeScript, it is strongly recommended to use `as const` assertion in
+combination with `satisfies SpritesheetMetadata`. This approach not only ensures compliance with the `SpritesheetMetadata`
+type but also enables autocompletion for animation names within utility functions, which is highly recommended:
+
+
+### `buildSpritesheet()`
+
+`buildSpritesheet()` is a utility function for building a final texture and spritesheet object from a provided `SpritesheetMetadata` object or external source.
+Each `buildSpritesheet` method return an Promise that and has to be awaited. Promise returned by each method contains an object with a `spritesheet`
+ready for use in `<InstancedSprite>`.
```

Comment:
You can just say the methods are async.

I have to say its a bit weird to me that you need to make a call to access some methods.

Have you considered either:
1. a simple object with methods
2. a `Spritesheet` class with a constructor that accepts `arg: string | SpritesheetMetadata` and instantiates accordingly